### PR TITLE
[FLINK-24167][Runtime]Add default HeartbeatReceiver and HeartbeatSend…

### PR DESCRIPTION
**Pull request by Aitozi**
…er to simplify the usage of HeartbeatTarget

## What is the purpose of the change

Add default HeartbeatReceiver and HeartbeatSender to simplify the usage of HeartbeatTarget

## Brief change log

Let most heartbeat listeners extend from `HeartbeatReceiver` or `HeartbeatSender`, instead of using `HeartbeatTarget` directly.

## Verifying this change

This change is already covered by existing tests

### CODE REVIEW

**Comment by Aitozi** [comment]
The failure case is related to [FLINK-24472](https://issues.apache.org/jira/browse/FLINK-24472) , rebase and run again.

File: flink-runtime/src/main/java/org/apache/flink/runtime/heartbeat/HeartbeatReceiver.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+package org.apache.flink.runtime.heartbeat;
+
+import org.apache.flink.runtime.clusterframework.types.ResourceID;
+import org.apache.flink.util.concurrent.FutureUtils;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * The receiver implementation of {@link HeartbeatTarget}, which mute the {@link
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/heartbeat/HeartbeatReceiver.java:27]:
```suggestion
 * The receiver implementation of {@link HeartbeatTarget}, which mutes the {@link
```

File: flink-runtime/src/main/java/org/apache/flink/runtime/heartbeat/HeartbeatReceiver.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * HeartbeatTarget#requestHeartbeat(ResourceID, I)}. The extender only have to care about the
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/heartbeat/HeartbeatReceiver.java:28]:
```suggestion
 * HeartbeatTarget#requestHeartbeat(ResourceID, I)}. The extender only has to care about the
```

**Review by tillrohrmann — APPROVED** [review]
Thanks for creating this PR @Aitozi. The changes look good to me. I will address my minor comments while merging this PR.

**Pull request closed without merging**